flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveSourceDynamicFileEnumerator.java

```diff
@@ -0,0 +1,209 @@
+        this.table = checkNotNull(table);
+        this.dynamicFilterPartitionKeys = checkNotNull(dynamicFilterPartitionKeys);
+        this.allPartitions = checkNotNull(allPartitions);
+        this.threadNum = threadNum;
+        this.jobConf = checkNotNull(jobConf);
+        this.defaultPartitionName = JobConfUtils.getDefaultPartitionName(jobConf);
+        this.hiveShim = HiveShimLoader.loadHiveShim(hiveVersion);
+        this.finalPartitions = this.allPartitions;
+    }
+
+    public void setDynamicFilteringData(DynamicFilteringData data) {
+        LOG.debug("Filtering partitions of table {} based on the data: {}", table, data);
+        if (!data.isFiltering()) {
+            finalPartitions = allPartitions;
+            return;
+        }
+        finalPartitions = new ArrayList<>();
+        RowType rowType = data.getRowType();
+        Preconditions.checkArgument(rowType.getFieldCount() == dynamicFilterPartitionKeys.size());
+        for (HiveTablePartition partition : allPartitions) {
+            RowData partitionRow = createRowData(rowType, partition.getPartitionSpec());
+            if (data.contains(partitionRow)) {
+                finalPartitions.add(partition);
+            }
+        }
+        LOG.info(
+                "Dynamic filtering table {}, original partition number is {}, remaining partition number {}",
+                table,
+                allPartitions.size(),
+                finalPartitions.size());
+    }
+
+    @VisibleForTesting
+    RowData createRowData(RowType rowType, Map<String, String> partitionSpec) {
+        GenericRowData rowData = new GenericRowData(rowType.getFieldCount());
+        for (int i = 0; i < rowType.getFieldCount(); ++i) {
+            String value = partitionSpec.get(dynamicFilterPartitionKeys.get(i));
+            Object convertedValue =
+                    HivePartitionUtils.restorePartitionValueFromType(
+                            hiveShim, value, rowType.getTypeAt(i), defaultPartitionName);
+            switch (rowType.getTypeAt(i).getTypeRoot()) {
+                case TINYINT:
+                case SMALLINT:
+                case INTEGER:
+                case BIGINT:
+                    // No further process is necessary.
+                    break;
+                case CHAR:
+                case VARCHAR:
+                    convertedValue = StringData.fromString((String) convertedValue);
+                    break;
+                case DATE:
+                    convertedValue =
+                            LocalDateConverter.INSTANCE.toInternal((LocalDate) convertedValue);
+                    break;
+                case TIMESTAMP_WITHOUT_TIME_ZONE:
+                    convertedValue =
+                            new LocalDateTimeConverter(9)
+                                    .toInternal((LocalDateTime) convertedValue);
+                    break;
+                default:
+                    if (DynamicFilteringData.SUPPORTED_TYPES.contains(
```

Comment:
Is it nessary to do such check? And is it expected all the other connectors implementing dynamic filter do such check?

Reply:
It's not necessary indeed, as it seems that supported types are relating to a specific connector type rather than the DynamicFilterData. I'll remove the check.